### PR TITLE
fix: make logbook accessible without much bugs

### PR DESCRIPTION
**Pull request by Junjiequan**
## Description
Few fixes:

1. fixed logbook data doesn't reload after first render. E.g., change tab causes logbook data missing.
2. fixed logbook not accessible from proposal page.
3. fixed logbook filter bar text search - when text search is empty nothing should happen. (previously, press Enter key on empty text search returns [Object Event])
4. fixed logbook router link is inactive when selecting filters.

## Motivation

As a user I want to see the logbook and use it without strange bugs!

## Fixes:

router link inactive bug:

https://user-images.githubusercontent.com/78078898/225615465-51d99ff1-97d1-4e73-b8ab-c05895b01559.mp4

## Tests included/Docs Updated?

- [ ] Included for each change/fix?
- [ ] Passing? (Merge will not be approved unless this is checked) 
- [ ] Docs updated?
- [ ] New packages used/requires npm install? 
- [ ] Toggle added for new features?
- [ ] Requires update of SciCat backend API?

### CODE REVIEW

**Review by martin-trajanovski — APPROVED** [review]
Just a small comment otherwise it looks good to me

File: src/app/proposals/view-proposal-page/view-proposal-page.component.ts [hunk 1]
```diff
@@ -102,11 +103,9 @@ export class ViewProposalPageComponent implements OnInit, OnDestroy {
       this.vm$.subscribe((vm) => {
         if (vm.proposal) {
           this.proposal = vm.proposal;
-          if (this.appConfig.logbookEnabled) {
```
[the remark this answers is not in the document]
Comment (nitrosx) [reply src/app/proposals/view-proposal-page/view-proposal-page.component.ts:105]: I agree.
If the logbook is not enabled, it should not even arrive here.

File: src/app/logbooks/logbooks-dashboard/logbooks-dashboard.component.html [hunk 2]
```diff
@@ -59,9 +59,77 @@
...
+      </div>
+    </div>
+  </div>
+
+  <ng-template [ngIf]="!(proposal_logbook.logbook | keyvalue)?.length ">
+    <mat-card class="no-logbook-message">
+      <mat-icon>info</mat-icon> No logbook available.
+    </mat-card>
+  </ng-template>
+</ng-container>
```
Comment (nitrosx) [inline src/app/logbooks/logbooks-dashboard/logbooks-dashboard.component.html:135]: Is it necessary to duplicate the whole element? Can we use the same element as we have under datasets?

Comment (Junjiequan) [reply src/app/logbooks/logbooks-dashboard/logbooks-dashboard.component.html:135]: I explained the reason on Slack. Its definitely not ideal, there's must be a cleaner way to address this, but I cant find better solution with Angular at the moment.

File: src/app/proposals/view-proposal-page/view-proposal-page.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Junjiequan) [reply src/app/proposals/view-proposal-page/view-proposal-page.component.ts:105]: Well, since you all agree its not necessary, I will just remove it.

**Review by nitrosx — APPROVED** [review]

**Pull request merged**